src/Kliva/Views/RecordPage.xaml.cs

```diff
@@ -41,14 +46,42 @@ private void OnViewModelPropertyChanged(object sender, System.ComponentModel.Pro
         {
             if (e.PropertyName.Equals(nameof(ViewModel.CurrentLocation), StringComparison.OrdinalIgnoreCase))
             {
-                TrackingMap.ClearMap<MapIcon>();
+                //ToDo PiNi: only do this the first time?
+                //and add button to re-center map?
+                //==> To discuss with team
+                CenterMap();
+
+                UpdateMapElements();
 
-                _currentLocation.Location = TrackingMap.Center = ViewModel.CurrentLocation;
-                TrackingMap.ZoomLevel = 16.0;
-                TrackingMap.MapElements.Add(_currentLocation);
+                if(!TrackingMap.MapElements.Contains(_currentLocation))
+                    TrackingMap.MapElements.Add(_currentLocation);
+
+                if (!TrackingMap.MapElements.Contains(_mapPolyline))
+                    TrackingMap.MapElements.Add(_mapPolyline);
             }
         }
 
+        private void UpdateMapElements()
+        {
+            List<BasicGeoposition> positions;
```

Comment:
Current solution is good!
But I didn't add it this way, because I was wondering if we can't let the GPX module of the Phone handle this?
There is a OnLocationServiceStatusChanged() method that you 'could' use in the RecordViewModel, this will be triggered each time there is a coordinate change ( need to uncomment _locationService.StatusChanged event handler in Constructor of RecordViewModel to activate ).
In theory this will contain the info of the change too...

So just wondering what would be best? For now merge PR and we can discuss later.